src/solace_ai_connector/components/general/openai/openai_chat_model_base.py

```diff
@@ -141,10 +143,22 @@ def invoke(self, message, data):
         if self.llm_mode == "stream":
             return self.invoke_stream(client, message, messages)
         else:
-            response = client.chat.completions.create(
-                messages=messages, model=self.model, temperature=self.temperature
-            )
-            return {"content": response.choices[0].message.content}
+            max_retries = 3
+            while max_retries > 0:
+                try:
+                    response = client.chat.completions.create(
+                        messages=messages,
+                        model=self.model,
+                        temperature=self.temperature,
+                    )
+                    return {"content": response.choices[0].message.content}
+                except Exception as e:
+                    log.error("Error invoking OpenAI: %s", e)
+                    max_retries -= 1
+                    if max_retries <= 0:
+                        raise e
+                    else:
+                        time.sleep(1)
```

Comment:
What's a scenario that the OpenAI would be down? If limit rate/out of credit, then sleep/retry will not fix the issue?

Reply:
We are seeing very frequent bedrock issues right now and there is also a quota reached error that might be fixed with this. Perhaps it won't be necessary in the future but with Bedrock it sadly is now